GiEnJul/Controllers/AdminController.cs

```diff
@@ -84,6 +84,15 @@ public async Task<FileStreamResult> DownloadExcelDeliveryLocationAsync(string lo
             using var wb = ExcelGenerator.Generate(_mapper.Map<IEnumerable<DeliveryExcel>>(connections));
             return wb.Deliver("leveranse_liste.xlsx");
         }
+        [HttpGet("completed/{location}")]
+        [Authorize(Policy = "ReadConnection")]
+        public async Task<IEnumerable<Entities.Connection>> GetConnectionsByLocation(string location)
```

Comment:
Controllers should not handle Entities. The repository function should return a Model-object, which then could create a DTO of the data you want to show.